cpp/CMakeLists.txt

```diff
@@ -0,0 +1,31 @@
+add_subdirectory(cluster)
+add_subdirectory(density)
+add_subdirectory(environment)
+add_subdirectory(locality)
+add_subdirectory(order)
+add_subdirectory(parallel)
+add_subdirectory(pmft)
+add_subdirectory(util)
+
+add_library(
+    libfreud
+    SHARED
+    $<TARGET_OBJECTS:_cluster>
+    $<TARGET_OBJECTS:_density>
+    $<TARGET_OBJECTS:_environment>
+    $<TARGET_OBJECTS:_locality>
+    $<TARGET_OBJECTS:_order>
+    $<TARGET_OBJECTS:_parallel>
+    $<TARGET_OBJECTS:_pmft>
+    $<TARGET_OBJECTS:_util>)
+
+target_link_libraries(
+    libfreud
+    PUBLIC TBB
+    )
+
+# Set the name of the library to just freud.
+set_target_properties(
+    libfreud
+    PROPERTIES OUTPUT_NAME freud
```

Comment:
It's not, however because it's a shared library and not a Python extension module CMake automatically prepends `lib` to the name. A `liblibfreud.so` seems very strange, and will also not be what people expect if we get around to exposing our C++ API. I don't want to name the actual library (in CMake) freud, though, because that conflicts with the project name and is confusing (not to mention I'm not 100% sure that CMake will always disambiguate correctly).

Reply:
@vyasr I would recommend clarifying this in the comment. I found that out earlier while playing with CMake and forgot that would happen. Maybe update the comment to say:
```
# CMake prepends 'lib' to the shared library name.
# This sets the name of the library to libfreud instead of liblibfreud.
```

Reply:
Good idea, I added more explicit documentation of this.

Reply:
Most old-style CMake scripts name the package in all caps to avoid the name conflict. Most new-style scripts use namespaces. You can name the target `freud::freud` and then the library should be named `libfreud` without needing `set_target_properties`

Reply:
It looks like that namespacing is not support for targets that must be built, only for imported ones: https://gitlab.kitware.com/cmake/cmake/-/issues/20444. So it works for TBB, but not for freud.